helper/inet/inet.go

```diff
@@ -0,0 +1,81 @@
+package inet
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+)
+
+const (
+	advertiseIp = iota
+	advertisePort
+)
+
+const (
+	address = iota
+	cidr
+)
+
+func AdvertiseIpFromSubnet(ip string) (string, error) {
+	// Return raw IP address if it is not in subnet-form
+	if !strings.Contains(ip, "/") {
+		return ip, nil
+	}
+
+	// Split subnet CIDR IP from the port
+	advertiseTokens := strings.Split(ip, ":")
+	if len(advertiseTokens) != 2 {
+		return "", errors.New(fmt.Sprintf("port was not given in advertise subnet: `%s`", ip))
+	}
+
+	// Split subnet IP and CIDR netmask by slash separator
+	ipTokens := strings.Split(advertiseTokens[advertiseIp], "/")
+	if len(ipTokens) != 2 {
+		return "", errors.New(fmt.Sprintf("cannot parse advertise subnet: `%s`", ip))
+	}
+
+	// Parse subnet IP and netmask
+	subnetIp := net.ParseIP(ipTokens[address])
+	if subnetIp == nil {
+		return "", errors.New(fmt.Sprintf("advertise subnet address out of range: `%s`", ipTokens[address]))
+	}
+	subnetCidr, err := strconv.ParseInt(ipTokens[cidr], 10, 32)
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("cannot parse advertise subnet CIDR: `%s`", ipTokens[cidr]))
+	}
+
+	subnet := net.IPNet{IP: subnetIp, Mask: net.CIDRMask(int(subnetCidr), 32)}
```

Comment:
So instead of doing all the work ourselves of splitting the CIDR string and parsing the parts, can we just use `net.ParseCIDR()`? It returns a net.IP and a *net.IPNet, which I think should get us the data we need.